R/getPhotoSearch.R

```diff
@@ -1,28 +1,103 @@
-#'
-#' @return This function returns data of specific photos matching search tags.
+#' @return This function returns data of specific photos matching search parameters.
 #' @export
 #'
 #' @examples
 #' \dontrun{
+#' # Search for photos tagged "cats"
+#' # Return images in descending order of date taken
+#'   getPhotoSearch(api_key = "XXXXXXXXXX",
+#'                sort = "date-taken-desc",
+#'                tags = "cats")
+#' }
+#' \dontrun{
+#' # Search for photos uploaded to the NPS Grand Canyon user account.
+#' # Return extra fields including the date taken and square image URL.
 #'   getPhotoSearch(api_key = "XXXXXXXXXX",
-#'                tags= "cats")
+#'                user_id = "grand_canyon_nps",
+#'                extras = c("date_taken", "url_sq"))
 #' }
-getPhotoSearch <- function(api_key, tags, page = 1, per_page = 500){
-  url = paste0("https://api.flickr.com/services/rest/?method=flickr.photos.search&api_key=",
-               api_key,
-               "&text=", paste(stringr::str_replace(tags, "@", "%40"), collapse = "%2C"),
-               "&per_page=", per_page,
-               "&page=", page,
-               "&format=json&nojsoncallback=1")
+#' #' \dontrun{
+#' # Search for photos tagged "panda" in the area of Ueno Zoo, Tokyo, Japan
+#' getPhotoSearch(api_key = "XXXXXXXXXX",
+#'                tags = "panda",
+#'                bbox = "139.7682226529,35.712627977,139.7724605432,35.7181464141",
+#'                extras = c("geo", "owner_name", "tags"))
+#' }
+#'
+#'
+
+getPhotoSearch <- function(api_key,
+                           user_id = NULL,
+                           tags = NULL,
+                           licence_id = NULL,
+                           sort = "date-posted-asc",
+                           bbox = NULL,
+                           extras,
+                           per_page = 100,
+                           page = 1) {
+  url = paste0(
+    "https://api.flickr.com/services/rest/?method=flickr.photos.search&api_key=",
+    api_key,
+    "&user_id=",
+    stringr::str_replace(user_id, "@", "%40"),
+    "&tags=",
+    paste(stringr::str_replace(tags, "@", "%40"), collapse = "%2C")
+  )
+
+  if (!missing(licence_id) && !(licence_id %in% c(0:10))) {
+    stop("The license id provided is not valid. The license id must be an integer from 0 to 10.")
+  }
+
+  url = paste0(url, "&license=", licence_id)
+
+  if (!(sort %in% c("date-posted-asc", "date-posted-desc", "date-taken-asc", "date-taken-desc", "interestingness-desc", "interestingness-asc", "relevance"))) {
+    stop("The sort provided is not valid. Possible sort values are: date-posted-asc, date-posted-desc, date-taken-asc, date-taken-desc, interestingness-desc, interestingness-asc, and relevance.")
+  }
+
+  url = paste0(url, "&sort=", sort)
+
+  if (class(bbox) == "bbox") {
+    bbox <- paste0(bbox[1], ",", bbox[2], ",", bbox[3], ",", bbox[4])
+  }
+
+  url = paste0(url, "&bbox=", bbox)
```

Comment:
selecting areas is really nice feature!!!!! i love it👍

but i actually encountered some errors when i set parameteres with spaces. here is the code which returned an error
```
getPhotoSearch(api_key = API_KEY,
               tags = "panda",
               bbox = "139.7682226529, 35.712627977, 139.7724605432, 35.7181464141",
               extras = c("geo", "owner_name", "tags"))
```
> error: lexical error: invalid char in json text.
                                       <!DOCTYPE HTML PUBLIC "-//IETF/
                     (right here) ------^ `

so here is my suggestion to remove all the scpaces from bbox input string

```url = paste0(url, "&bbox=", stringr::str_remove_all(bbox, "\\s"))```

Reply:
On reflection, I wonder if it makes sense to require a vector of coordinates instead of allowing strings at all. This examples would be changed to look like:

`getPhotoSearch(api_key = API_KEY,
               tags = "panda",
               bbox = c(139.7682226529, 35.712627977, 139.7724605432, 35.7181464141),
               extras = c("geo", "owner_name", "tags"))`

The same code used to turn a bbox class object into the appropriate string for the Flickr API call would work for this vector xmin, ymin, xmax, ymax format just as well.